R/AzureDataLake.R

```diff
@@ -0,0 +1,368 @@
+
+  if (!missing(azureActiveContext) && !is.null(azureActiveContext)) {
+    assert_that(is.azureActiveContext(azureActiveContext))
+    azureCheckToken(azureActiveContext)
+  }
+  assert_that(is_storage_account(azureDataLakeAccount))
+  verbosity <- set_verbosity(verbose)
+
+  URL <- paste0(
+    "https://", azureDataLakeAccount, ".azuredatalakestore.net/webhdfs/v1/",
+    relativePath,
+    "?op=LISTSTATUS",
+    "&api-version=2016-11-01"
+    )
+
+  resHttp <- callAzureDataLakeApi(URL,
+    azureActiveContext = azureActiveContext,
+    verbose = verbose)
+
+  stopWithAzureError(resHttp)
+
+  resJsonStr <- content(resHttp, "text", encoding = "UTF-8")
+  resJsonObj <- jsonlite::fromJSON(resJsonStr)
+  if (length(resJsonObj$FileStatuses$FileStatus) == 0) {
+    #return empty data frame in case of an empty json object
+    return(
+      data.frame(
+        FileStatuses.FileStatus.length = character(0),
+        FileStatuses.FileStatus.pathSuffix = character(0),
+        FileStatuses.FileStatus.type = character(0),
+        FileStatuses.FileStatus.blockSize = character(0),
+        FileStatuses.FileStatus.accessTime = character(0),
+        FileStatuses.FileStatus.modificationTime = character(0),
+        FileStatuses.FileStatus.replication = character(0),
+        FileStatuses.FileStatus.permission = character(0),
+        FileStatuses.FileStatus.owner = character(0),
+        FileStatuses.FileStatus.group = character(0),
+        FileStatuses.FileStatus.aclBit = character(0)
+      )
+    )
+  }
+  resDf <- as.data.frame(resJsonObj)
+  resDf
+}
+
+
+#' Azure Data Lake GETFILESTATUS for specified relativePath of an azure data lake account.
+#'
+#' @inheritParams createAzureContext
+#' @inheritParams azureAuthenticate
+#'
+#' @param azureActiveContext Provide an `azureActiveContext` object used for authentication.
+#' @param azureDataLakeAccount Provide the name of the Azure Data Lake account.
+#' @param relativePath Provide a relative path of the directory.
+#' @param verbose Print tracing information (default FALSE).
+#'
+#' @return Returns a data frame.
+#'
+#' @template
+#' @references
+#' @family Azure Data Lake Store functions
+#' @export
+azureDataLakeGetFileStatus <- function(azureActiveContext, azureDataLakeAccount, relativePath = "", verbose = FALSE) {
+  
+  if (!missing(azureActiveContext) && !is.null(azureActiveContext)) {
+    assert_that(is.azureActiveContext(azureActiveContext))
+    azureCheckToken(azureActiveContext)
+  }
+  assert_that(is_storage_account(azureDataLakeAccount))
+  verbosity <- set_verbosity(verbose)
+  
+  URL <- paste0(
+    "https://", azureDataLakeAccount, ".azuredatalakestore.net/webhdfs/v1/",
+    relativePath,
+    "?op=GETFILESTATUS",
+    "&api-version=2016-11-01"
+  )
+  
+  resHttp <- callAzureDataLakeApi(URL,
+                                  azureActiveContext = azureActiveContext,
+                                  verbose = verbose)
+  
+  if (status_code(resHttp) == 404) {
```

Comment:
Error handling seems inconsistent and arbitrary. If this is worth doing here, isn't it worth doing for azureDataLakeListStatus ?

Reply:
Fixed in https://github.com/omkarksa/AzureSMR/pull/1 will raise a pull request once this is merged into master